### PR TITLE
Performance improvements

### PR DESCRIPTION
**Pull request by russellhancox**
Mostly perf improvements:

* Use msleep/wakeup instead of sleeping for a fixed period while waiting for responses from daemon
* Cache user/group id->name lookups as getpwuid/getgrgid can be quite slow
* Rename the Dismiss button to Ignore

### CODE REVIEW

File: Source/santad/SNTEventLog.m [hunk 1]
```diff
@@ -29,6 +29,10 @@
 @interface SNTEventLog ()
 @property NSMutableDictionary *detailStore;
 @property dispatch_queue_t detailStoreQueue;
+
+@property NSMutableDictionary *userNameMap;
```
Comment (tburgin) [inline Source/santad/SNTEventLog.m]: Add a comment to userNameMap and groupNameMap. Something like:
Do not access directly, this is used for caching. Only access from the nameMapQueue.

File: Source/santad/SNTExecutionController.m [hunk 2]
```diff
@@ -365,44 +373,36 @@ - (void)printMessage:(NSString *)msg toTTYForPID:(pid_t)pid {
     return;
   }
 
-  NSString *devPath = [NSString stringWithFormat:@"/dev/%s", devname(taskInfo.e_tdev, S_IFCHR)];
-  int fd = open(devPath.UTF8String, O_WRONLY | O_NOCTTY);
-  @try {
-    NSFileHandle *fh = [[NSFileHandle alloc] initWithFileDescriptor:fd closeOnDealloc:YES];
-    [fh writeData:[msg dataUsingEncoding:NSUTF8StringEncoding]];
-  } @catch (NSException *) { /* do nothing */ }
+  char devPath[32] = "/dev/";
```
Comment (tburgin) [inline Source/santad/SNTExecutionController.m]: 32 seems arbitrary. Either use `PATH_MAX` or go the other way and count the max tty sessions. `kern.tty.ptmx_max` seems to take a max of 999. So the max "/dev/tty999" is 12 bytes with null termination.

File: Source/santad/SNTEventLog.m (hunk 1 above)
Comment (russellhancox) [inline Source/santad/SNTEventLog.m]: Added a comment

File: Source/santad/SNTExecutionController.m (hunk 2 above)
Comment (russellhancox) [inline Source/santad/SNTExecutionController.m]: Added a comment

**Comment by tburgin** [comment]
LGTM

**Pull request merged**